src/test/java/org/mockitousage/matchers/CapturingArgumentsTest.java

```diff
@@ -298,7 +313,7 @@ public void captures_correctly_when_captor_used_multiple_times() throws Exceptio
 
         // then
         // this is only for backwards compatibility. It does not make sense in real to do so.
-        verify(mock).mixedVarargs(any(), argumentCaptor.capture(), argumentCaptor.capture(), argumentCaptor.capture());
+        verify(mock).mixedVarargs(any(), argumentCaptor.capture());
```

Comment:
@terebesirobert Thanks for pointing me to the problematic code! 
As far as I can see this loop in `captureVarargsPart(Invocation invocation)` is responsible for adding every vararg to every AgrumentMatcher, which leads to our failing test:
```
for (int i = 0; i < Array.getLength(rawArgument); i++) {
  ((CapturesArguments) m).captureFrom(Array.get(rawArgument, i));
}
```
Do you see a chance to fix the problem right here? Maybe we don't need to add eq/hashCode to every matcher.